src/api/java/io/github/opencubicchunks/cubicchunks/api/world/ICube.java

```diff
@@ -250,4 +252,6 @@
     boolean isCubeLoaded();
 
     boolean hasLightUpdates();
+
+    Biome getBiome(BlockPos pos, BiomeProvider provider);
```

Comment:
Why the biome provider here?

Reply:
To call chunk 'getBiome(BlockPos, BiomeProvider)' method if blockBiomeArray is null.

Reply:
Looks like Chunk does the same. Seems a bit stupid for such rare case. One of the things I would suggest to change from vanilla and just get the BiomeProvider from world.